src/Abp.Zero.Common/Organizations/OrganizationUnitManager.cs

```diff
@@ -77,14 +77,16 @@ public virtual string GetNextChildCode(long? parentId)
 
         public virtual async Task<OrganizationUnit> GetLastChildOrNullAsync(long? parentId)
         {
-            var children = await OrganizationUnitRepository.GetAllListAsync(ou => ou.ParentId == parentId);
-            return children.OrderBy(c => c.Code).LastOrDefault();
+            return await OrganizationUnitRepository.GetAll()
+                .OrderByDescending(c => c.Code)
+                .FirstOrDefaultAsync(ou => ou.ParentId == parentId);
```

Comment:
@PMExtra your change requires adding EntityFramework reference to Abp.Zero.Common project but we don't wnat that. My suggestion below should work. Could you change it and update the PR ?

```suggestion
            var lastChild = OrganizationUnitRepository.GetAll()
                .OrderByDescending(c => c.Code)
                .FirstOrDefault(ou => ou.ParentId == parentId);

            return await Task.FromResult(lastChild);
```

Reply:
I thought it's not awesome because it's not real async.
Shall we consider extending `IRepository`?

Reply:
Right but this project is common for EF and EF Core, so we can't add them together.
How can we extend IRepository ?

Reply:
Let’s use `IAsyncQuerableExecutor` ?

Reply:
@ryancyq Could you tell me more detail?

Reply:
See https://github.com/aspnetboilerplate/aspnetboilerplate/blob/945cc80c238cbdd4d9821ef99d12fe6c9b6e33df/src/Abp.Zero.Common/WebHooks/WebHookSubscriptionsStore.cs#L83

`AsyncQueryableExecuter` provides Async Linq abstraction

Reply:
Perfect!